src/main/java/io/ifar/archive/core/ArchiveListener.java

```diff
@@ -48,11 +56,43 @@ public ArchiveListener(AmazonS3Client s3Client, String seedBrokers, String workU
         this.executor = Executors.newScheduledThreadPool(maxNumParallelWorkers);
         this.metrics = metrics;
         this.topicConfiguration = topicConfiguration;
+        this.errorLimitConfiguration = errorLimitConfiguration;
         this.defaultBucket = defaultBucket;
 
         numWorkUnitsGauge = metrics.register("archiveListener-workUnits", new IntegerGauge());
     }
 
+    public void setCluster(ArchiveCluster archiveCluster) {
+        this.archiveCluster = archiveCluster;
+    }
+
+    public void batchFailure(String workUnit) {
+        if (!stopFlag.get()) {
+            int errors = failedBatches.getAndIncrement()+1;
+            LOG.warn("Encountered error in archive worker {} ({} error(s) total)", workUnit, errors);
+            int successful = successfulBatches.get();
+
+            if (errors >= errorLimitConfiguration.getTriesBeforeEnableErrorLimit()) {
+
+                double limit = errorLimitConfiguration.getErrorLimitPercent() / 100.0;
+
+                if ( (1.0*errors) / (1.0*successful) >= limit ) {
+                    LOG.error("Error limit reached!");
```

Comment:
It would make sense to add bit more information here I think, on number of errors/successful batches, and perhaps mention what is being tried (going to stop and wait on cluster listener)? Mostly because while intent is clear now, in a few months when problem resurfaces, it's helpful to get bit more context to go back to figuring out what is going on.